exercise_fruit_processor

```diff
@@ -0,0 +1,119 @@
+# create a fruit processor to process Apple and Banana, requirement is below:
+# 1. they all have a property like color, weight
+# 2. they all have method Process() and output is AppleJuice and BananaJuice and BlendJuice
+# 3. they need to be initialized in the very beginning
+# 4. there is a input for different fruits, but need to output a juice with
+# 4.1 different weight (sum of existing fruit)
+# 4.2 different color (red, yellow or blending)
+#
+#
+#
+class Fruit
+
+  attr_reader :name, :color, :weight
+
+  def name=(name)
+    if name == ""
+      raise "fruit name cannot be blank."
+    else
+      @name = name
+    end
+  end
+
+  def color=(color)
+    if color == ""
+      raise "fruit color cannot be blank."
+    else
+      @color = color
+    end
+  end
+
+  def weight=(weight)
+    if weight < 0
+      raise "value of weight is invalid."
+    else
+      @weight = weight
+    end
+  end
+
+  def initialize(name, color, weight)
+    self.name = name
+    self.color = color
+    self.weight = weight
+  end
+
+  def print_fruit_info
+    puts "Fruit Name: #{name}"
+    puts "The fruit is '#{color}' color and weight is #{weight}."
+  end
+end
+
+class JuiceMachine
+
+  attr_accessor :fruit, :juice, :color, :weight
+
+  def initialize(fruit, juice)
+    self.fruit = fruit
+    self.juice = juice
+  end
+
+  def process
+
+    @weight = juice.weight + fruit.weight
+
+    if juice.color == fruit.color
+      @color = fruit.color
+    else
+      @color = "blending"
+    end
+
+  end
+
+  def print_juice
+    puts "Juice is coming ...  (wait for 3 seconds)"
+    puts "#{@weight}g #{@color} juice!"
+  end
+
+end
+
+puts "Welcome to use 'Juice Machine'!"
+puts "please put fruit(name color weight) into the juice machine (input `done` to complete it) "
+
+more_fruit = true
+fruit_number = 0
+
+while more_fruit
+
+  print "Input fruit: "
+  name, color, weight = gets.chomp.split
+  weight = weight.to_i
+  #puts "Name: #{name}, color:#{color}, weight: #{weight}"
+
+  if name == "done"
+    more_fruit = false
+  else
+    fruit = Fruit.new(name, color, weight)
+    fruit_number += 1
+    if fruit_number == 1
+      juice_weight = fruit.weight
+      juice_color = fruit.color
+      juice = fruit
+    else
+      juice = JuiceMachine.new(fruit, juice)
+      juice.process
+    end
+  end
+
+end
+
+unless more_fruit
+
+  if fruit_number == 1
+    puts "Juice is coming ...  (wait for 3 seconds)"
+    puts "#{juice_weight}g #{juice_color} juice!"
+  else
+    juice.print_juice
+  end
+
+end
```

Comment:
1. try to see whether can avoid the duplication of `Juice is coming ...  (wait for 3 seconds)`
2. both `while` and `unless` operate the `more_fruit`
3. can you as more as you can the code into JuiceMachine?  it will looks cool.